statemachines/pre_flight_checks/test_system_requirements

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+DIR=`dirname $0`
+. /etc/restatemachine/shell_helpers.inc
+
+INPUT=`cat`
+Error () {
+        echo -n "$INPUT"
+        restatemachine_transition_to "test_system_requirements"
+        restatemachine_transition_after_seconds "60"
+        restatemachine_status_message "$1"
+        exit 0
+}
+
+HOSTNAME=`echo "$INPUT" | jq -r '.hostname'`
+USERNAME=`echo "$INPUT" | jq -r '.username'`
+PASSWORD=`echo "$INPUT" | jq -r '.password'`
+KEY=`echo "$INPUT" | jq -r '.key'`
+OS=`echo "$INPUT" | jq -r '.os'`
+CPU=`echo "$INPUT" | jq -r '.system_requirements.cpu'`
+RAM=`echo "$INPUT" | jq -r '.system_requirements.ram'`
+DISK=`echo "$INPUT" | jq -r '.system_requirements.disk'`
+
+MESSAGE="{\"status\":\"failed\""
+if [ "$OS" == "linux" ]; then
+	if [ "$KEY" != "" ]; then
+		AVAILABLE_RAM=`ssh -oStrictHostKeyChecking=no -i "$KEY" "$USERNAME"@"$HOSTNAME" -q 'sudo dmidecode -t 17 2>&1 | awk '"'"'( /Size/ && $2 ~ /^[0-9]+$/ ) { x+=$2 } END{ print x }'"'"''`
+		if [ "$?" != "0" ]; then
+			Error "{\"status\":\"failed\", \"message\":\"Could not verify RAM requirement\"}"
+		fi
+
+		AVAILABLE_CPU=`ssh -oStrictHostKeyChecking=no -i "$KEY" "$USERNAME"@"$HOSTNAME" -q ' grep -c '"'"'^processor'"'"' /proc/cpuinfo 2>&1'`
+		if [ "$?" != "0" ]; then
+			Error "{\"status\":\"failed\", \"message\":\"Could not verify CPU requirement\"}"
+		fi
+
+		AVAILABLE_DISK=`ssh -oStrictHostKeyChecking=no -i "$KEY" "$USERNAME"@"$HOSTNAME" -q 'sudo df -h 2>&1| egrep '"'"'/$'"'"' | awk '"'"'{ print $2 }'"'"' | sed s/G//'`
+		if [ "$?" != "0" ]; then
+			Error "{\"status\":\"failed\", \"message\":\"Could not verify DISK requirement\"}"
+		fi
+
+	else
+		AVAILABLE_RAM=`sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$USERNAME"@"$HOSTNAME" -q 'sudo dmidecode -t 17 2>&1 | awk '"'"'( /Size/ && $2 ~ /^[0-9]+$/ ) { x+=$2 } END{ print x }'"'"''`
+		if [ "$?" != "0" ]; then
+			Error "{\"status\":\"failed\", \"message\":\"Could not verify RAM requirement\"}"
+		fi
+                AVAILABLE_CPU=`sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$USERNAME"@"$HOSTNAME" -q ' grep -c '"'"'^processor'"'"' /proc/cpuinfo 2>&1'`
+                if [ "$?" != "0" ]; then
+                        Error "{\"status\":\"failed\", \"message\":\"Could not verify CPU requirement\"}"
+                fi
+
+                AVAILABLE_DISK=`sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$USERNAME"@"$HOSTNAME" -q 'sudo df -h 2>&1| egrep '"'"'/$'"'"' | awk '"'"'{ print $2 }'"'"' | sed s/G//'`
+                if [ "$?" != "0" ]; then
+                        Error "{\"status\":\"failed\", \"message\":\"Could not verify DISK requirement\"}"
+                fi
+	fi
+
+	HAS_ERRORS=0
+	if [[ $AVAILABLE_RAM -lt $RAM ]]; then
```

Comment:
Never do this, always "$AVAILABLE_RAM" if you don't explicitly want to expand shell variables etc, so that you only break the flow with funky values instead of getting shell injection problems or other nasty things.